### PR TITLE
Feature: Introduce orb filename parameter

### PR DESCRIPTION
**Pull request by adomaskizogian**
Introduce new parameters on pack and publish jobs to allow setting custom pack output filename.

To avoid breaking changes these parameters default to previously hardcoded value on scripts - `orb.yml`

Motivation for this change is that running multiple pack/publish jobs for different orbs within a singlw workflow does not work. When multiple orbs are packed and published, run on circleci fails as multiple files with the same name are attached on the workspace resulting in conflicts in the form of: 
```
Concurrent upstream jobs persisted the same file(s) into the workspace:
  - orb.yml
```

In other words, this proposal can be seen as a bugfix.

In our use case we have set up a monorepo containing source for multiple orbs. On pull-request merges to master all orbs are published as dev versions. Releases - publish version - are done for each orb individually - using version tags prefixed with the orb name.

### CODE REVIEW

**Comment by KyleTryon** [comment]
Thank you for your submission @adomaskizogian!

To ensure we get the latest version of orb tools out in a timely manner, I have wrapped up your suggestion into the current Orb Tools 12 PR. 

This does mean unfortunately we will not be able to add this to the 11.x release, mostly due to time constraints and some major changes we are adopting to naming conventions moving forward. 
I have listed your PR in the notes here: https://github.com/CircleCI-Public/orb-tools-orb/pull/181

Please feel free to review, your changes have already been included, thank you again.
